quickwit/quickwit-proto/src/lib.rs

```diff
@@ -341,6 +344,49 @@ impl TryFrom<&str> for IndexingTask {
     }
 }
 
+
+/// Creates a query ast json by parsing a user query.
+///
+/// The resulting query does not include `UserInputQuery` nodes.
+/// The resolution assumes that there are no default search fields
+/// in the doc mapper.
+///
+/// # Panics
+///
+/// Panics if the user text is invalid.
+pub fn qast_helper(
+    user_text: &str,
+    default_fields: &[&'static str]
+) -> String {
+    let default_fields: Vec<String> = default_fields.iter().map(|default_field| default_field.to_string()).collect();
+    let ast: QueryAst = query_ast_from_user_text(user_text, Some(default_fields))
+        .parse_user_query(&[])
+        .expect("Invalid user query");
+    serde_json::to_string(&ast).expect("Failed to serialize ast")
+}
+
+/// Creates a QueryAST with a single UserInputQuery node.
+///
+/// Disclaimer:
+/// At this point the query has not been parsed.
+///
+/// The actual parsing is meant to happen on a root node,
+/// `default_fields` can be passed to decide which field should be search
+/// if not specified specifically in the user query (e.g. hello as opposed to "body:hello").
+///
+/// If it is not supplied, the docmapper search fields are meant to be used.
+pub fn query_ast_from_user_text(
+    user_text: &str,
+    default_fields: Option<Vec<String>>,
+) -> QueryAst {
+    quickwit_query::query_ast::UserInputQuery {
+        user_text: user_text.to_string(),
+        default_fields,
+        default_operator: quickwit_query::DefaultOperator::Or,
```

Comment:
I'm surprised by the default "OR", I thought we went for "AND".

Reply:
Ah yes! It was done for elastic, but I can probably have a different behavior for elastic and quickwit.